neurons/validator.py

```diff
@@ -203,6 +210,106 @@ async def competition_loop_tick(self):
         ]
         self.save_state()
 
+    async def fetch_organization_data_references(self):
+        api = HfApi()
+
+        files = api.list_repo_tree(
+            repo_id=self.config.datasets_config_hf_repo_id,
+            repo_type="space",
+            token=self.config.hf_token,
+            recursive=True,
+            expand=True,
+        )
+
+        yaml_files = []
+
+        for file_info in files:
+            if file_info.__class__.__name__ == 'RepoFile':
+                file_path = file_info.path
+
+                if file_path.startswith('datasets/') and file_path.endswith('.yaml'):
+                    local_file_path = hf_hub_download(
+                        repo_id=self.config.datasets_config_hf_repo_id,
+                        repo_type="space",
+                        token=self.config.hf_token,
+                        filename=file_path,
+                    )
+
+                    last_commit_info = file_info.last_commit
+                    commit_date = last_commit_info.date if last_commit_info else None
+
+                    if commit_date is not None:
+                        date_uploaded = commit_date
+                    else:
+                        bt.logging.warning(f"Could not get the last commit date for {file_path}")
+                        date_uploaded = None
+
+                    with open(local_file_path, 'r') as f:
+                        yaml_data = yaml.safe_load(f)
+
+                    yaml_files.append({
+                        'file_name': file_path,
+                        'yaml_data': yaml_data,
+                        'date_uploaded': date_uploaded
+                    })
+            else:
+                continue
+        return yaml_files
```

Comment:
why you don't you create objects right away? 
anyway, your are not returning "yaml_files", but some objects right? it's better to return ready data for further processing

Also typings

Reply:
Yeah I had some separating concerns and wanted to have separate function just for fetching logic/possible errors. I like the idea of single function -> single responsibility approach. Anyway I agree that it shouldnt be called yaml_files. I hesitated with a separate structure because the models for mapping all that data already exist and this is just pre-parsing to make it easier for the other function to handle the logic related to updating the singleton class and catch the possible update